### PR TITLE
TTINTEL-291

### PR DESCRIPTION
**Pull request by jnunyez**

### CODE REVIEW

File: src/vse_sync_pp/analyzers/phc2sys.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Analyze ts2phc log messages"""
```
Comment (spenced) [inline src/vse_sync_pp/analyzers/phc2sys.py:3]:
```suggestion
"""Analyze phc2sys log messages"""
```

File: src/vse_sync_pp/parsers/phc2sys.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Parse ts2phc log messages"""
```
Comment (spenced) [inline src/vse_sync_pp/parsers/phc2sys.py:3]:
```suggestion
"""Parse phc2sys log messages"""
```

File: src/vse_sync_pp/parsers/phc2sys.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import re
+from collections import namedtuple
+
+from .parser import (Parser, parse_decimal)
+
+class TimeErrorParser(Parser):
+    """Parse time error from a ts2phc log message"""
```
Comment (spenced) [inline src/vse_sync_pp/parsers/phc2sys.py:11]:
```suggestion
    """Parse time error from a phc2sys log message"""
```

File: src/vse_sync_pp/requirements.py [hunk 4]
```diff
@@ -9,4 +9,7 @@
     'G.8272/PRTC-B': {
         'time-error-in-locked-mode/ns': 40,
     },
+    'workload/RAN': {
```
Comment (spenced) [inline src/vse_sync_pp/requirements.py:12]: Where did the name come from?

File: src/vse_sync_pp/requirements.py [hunk 5]
```diff
@@ -9,4 +9,7 @@
...
+        'time-error-in-locked-mode/ns': 100,
```
Comment (spenced) [inline src/vse_sync_pp/requirements.py:13]: Where did the value come from?

File: tests/vse_sync_pp/analyzers/test_phc2sys.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Test cases for vse_sync_pp.analyzers.ts2phc"""
```
Comment (spenced) [inline tests/vse_sync_pp/analyzers/test_phc2sys.py:3]:
```suggestion
"""Test cases for vse_sync_pp.analyzers.phc2sys"""
```

File: tests/vse_sync_pp/analyzers/test_phc2sys.py [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+from vse_sync_pp.analyzers.phc2sys import (
+    TimeErrorAnalyzer
+)
+
+from .test_analyzer import AnalyzerTestBuilder
+
+TERR = namedtuple('TERR', ('timestamp', 'terror', 'state', 'delay'))
+
+class TestTimeErrorAnalyzer(TestCase, metaclass=AnalyzerTestBuilder):
+    """Test cases for vse_sync_pp.analyzers.ts2phc.TimeErrorAnalyzer"""
```
Comment (spenced) [inline tests/vse_sync_pp/analyzers/test_phc2sys.py:19]:
```suggestion
    """Test cases for vse_sync_pp.analyzers.phc2sys.TimeErrorAnalyzer"""
```

File: tests/vse_sync_pp/parsers/test_phc2sys.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Test cases for vse_sync_pp.parsers.ts2phc"""
```
Comment (spenced) [inline tests/vse_sync_pp/parsers/test_phc2sys.py:3]:
```suggestion
"""Test cases for vse_sync_pp.parsers.phc2sys"""
```

File: tests/vse_sync_pp/parsers/test_phc2sys.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+from decimal import Decimal
+
+from vse_sync_pp.parsers.phc2sys import (
+    TimeErrorParser,
+)
+
+from .test_parser import ParserTestBuilder
+
+class TestTimeErrorParser(TestCase, metaclass=ParserTestBuilder):
+    """Test cases for vse_sync_pp.parsers.ts2phc.TimeErrorParser"""
```
Comment (spenced) [inline tests/vse_sync_pp/parsers/test_phc2sys.py:15]:
```suggestion
    """Test cases for vse_sync_pp.parsers.phc2sys.TimeErrorParser"""
```

**Review by spenced — CHANGES_REQUESTED** [review]

File: src/vse_sync_pp/parse.py [hunk 10]
```diff
@@ -11,6 +11,7 @@
     JsonEncoder,
 )
 
+
```
Comment (spenced) [inline src/vse_sync_pp/parse.py:14]:
```suggestion
```

File: tests/vse_sync_pp/analyzers/test_phc2sys.py (hunk 6 above)
Comment (spenced) [inline tests/vse_sync_pp/analyzers/test_phc2sys.py:3]:
```suggestion
"""Test cases for vse_sync_pp.analyzers.phc2sys"""
```

File: tests/vse_sync_pp/analyzers/test_phc2sys.py [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+"""Test cases for vse_sync_pp.analyzers.ts2phc"""
+
+from unittest import TestCase
+from collections import namedtuple
+from decimal import Decimal
+import math
+
+from vse_sync_pp.analyzers.phc2sys import (
+    TimeErrorAnalyzer
+)
```
Comment (spenced) [inline tests/vse_sync_pp/analyzers/test_phc2sys.py:12]:
```suggestion
from vse_sync_pp.analyzers.phc2sys import TimeErrorAnalyzer
```

Comment (spenced) [reply tests/vse_sync_pp/analyzers/test_phc2sys.py:12]: Alternatively, add a comma after `TimeErrorAnalyzer` if you're listing symbols out

File: tests/vse_sync_pp/analyzers/test_phc2sys.py (hunk 7 above)
Comment (spenced) [inline tests/vse_sync_pp/analyzers/test_phc2sys.py:19]:
```suggestion
    """Test cases for vse_sync_pp.analyzers.phc2sys.TimeErrorAnalyzer"""
```

File: tests/vse_sync_pp/parsers/test_phc2sys.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        '\n'.join((
+            'foo',
+            'phc2sys[847914.839]: [ptp4l.0.config] '
+            'CLOCK_REALTIME phc offset          1 s2 freq      +639 delay    102',
+            'bar',
+            'phc2sys[847915.839]: [ptp4l.0.config] '
+            'CLOCK_REALTIME phc offset          0 s2 freq      +639 delay    102',
+            'baz',
+            'phc2sys[847916.839]: '
+            'CLOCK_REALTIME phc offset          4 s2 freq      +639 delay    102',
```
Comment (nocturnalastro) [inline tests/vse_sync_pp/parsers/test_phc2sys.py:39]: Should lined 38 & 39 be their own test in accept?

File: tests/vse_sync_pp/parsers/test_phc2sys.py [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+from .test_parser import ParserTestBuilder
+
+class TestTimeErrorParser(TestCase, metaclass=ParserTestBuilder):
+    """Test cases for vse_sync_pp.parsers.phc2sys.TimeErrorParser"""
+    constructor = TimeErrorParser
+    id_ = 'phc2sys/time-error'
+    elems = ('timestamp', 'terror', 'state', 'delay')
+    accept = (
+        (   'phc2sys[681011.839]: [ptp4l.0.config] '
```
Comment (spenced) [inline tests/vse_sync_pp/parsers/test_phc2sys.py:20]: Also need a case where ` [ptp4l.0.config]` (including leading single space) is absent

File: tests/vse_sync_pp/parsers/test_phc2sys.py [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    file = (
+        '\n'.join((
+            'foo',
+            'phc2sys[847914.839]: [ptp4l.0.config] '
+            'CLOCK_REALTIME phc offset          1 s2 freq      +639 delay    102',
+            'bar',
+            'phc2sys[847915.839]: [ptp4l.0.config] '
+            'CLOCK_REALTIME phc offset          0 s2 freq      +639 delay    102',
+            'baz',
+            'phc2sys[847916.839]: '
```
Comment (spenced) [inline tests/vse_sync_pp/parsers/test_phc2sys.py:38]: Acknowledging that config file item is absent here (I still propose it needs to be in `accept`).

**Review by spenced — CHANGES_REQUESTED** [review]
A few small comments, looks in good shape.

File: tests/vse_sync_pp/parsers/test_phc2sys.py (hunk 12 above)
Comment (nocturnalastro) [reply tests/vse_sync_pp/parsers/test_phc2sys.py:39]: Already covered in spenced's comment

File: src/vse_sync_pp/parsers/phc2sys.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+"""Parse phc2sys log messages"""
+
+import re
+from collections import namedtuple
+
+from .parser import (Parser, parse_decimal)
+
+class TimeErrorParser(Parser):
+    """Parse time error from a p log message"""
```
Comment (spenced) [inline src/vse_sync_pp/parsers/phc2sys.py:11]:
```suggestion
    """Parse time error from a phc2sys log message"""
```

**Review by spenced — APPROVED** [review]

**Review by nocturnalastro — APPROVED** [review]

**Pull request merged**